plugins/modules/backup_tag.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r"""
+---
+module: backup_tag
+version_added: 6.0.0
+short_description: Manage tags on backup plan, backup vault, recovery point.
+description:
+    - Create, list, update, remove tags on AWS backup resources such as backup plan, backup vault, and recovery point.
+    - Resources are referenced using ARN.
+author:
+    - Mandar Vijay Kulkarni (@mandar242)
+options:
+  resource:
+    description:
+      - The Amazon Resource Name (ARN) of the backup resource.
+    required: true
+    type: str
+  state:
+    description:
+      - Whether the tags should be present or absent on the resource.
+    default: present
+    choices: ['present', 'absent']
+    type: str
+  tags:
+    description:
+      - A dictionary of tags to add or remove from the resource.
+      - If the value provided for a tag key is null and I(state=absent), the tag will be removed regardless of its current value.
+    type: dict
+    required: true
+    aliases: ['resource_tags']
+  purge_tags:
+    description:
+      - Whether unspecified tags should be removed from the resource.
+      - Note that when combined with I(state=absent), specified tag keys are not purged regardless of its current value.
+    type: bool
+    default: false
+
+extends_documentation_fragment:
+  - amazon.aws.common.modules
+  - amazon.aws.region.modules
+  - amazon.aws.tags
```

Comment:
```suggestion
```

tags and purge tags are both overwridden due to the subtly different logic.

What might be nice would be to create a separate fragment which all of our `_tag` modules could share.